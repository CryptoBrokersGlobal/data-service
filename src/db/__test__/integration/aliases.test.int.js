const { Nothing } = require('folktale/maybe');

const loadConfig = require('../../../loadConfig');
const createDb = require('../../index');

const db = createDb(loadConfig());

describe('Aliases should return', () => {
  it('Maybe(alias) for `one` correctly', done => {
    db.aliases
      .one('sexy-boys')
      .run()
      .listen({
        onResolved: maybeX => {
          expect(maybeX).toMatchSnapshot();
          done();
        },
      });

    db.aliases
      .one('NON_EXISTING_ALIAS')
      .run()
      .listen({
        onResolved: maybeX => {
          expect(maybeX).toEqual(Nothing());
          done();
        },
      });
  });

  it('Maybe(data)[] for `many` aliases request', done => {
    db.aliases
      .many({ address: '3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM' })
      .run()
      .listen({
        onResolved: mxs => {
          const xs = mxs.map(x => x.getOrElse(-1));
          expect(xs).toMatchSnapshot();
          done();
        },
      });
  });
});
