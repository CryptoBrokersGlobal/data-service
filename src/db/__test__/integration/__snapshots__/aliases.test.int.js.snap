// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aliases should return Maybe(alias) for \`one\` correctly 1`] = `
Object {
  "@@tag": "Just",
  "@@type": "folktale:Maybe",
  "@@value": Object {
    "value": anonymous {
      "address": "3PQqCXNPoTRsHRK2nZyXWENaZwLmxrUZPBi",
      "alias": "sexy-boys",
    },
  },
}
`;

exports[`Aliases should return Maybe(data)[] for \`many\` aliases request 1`] = `
Array [
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "jeffrey",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "hans",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "hansch",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "liqio.io",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "trextracker",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "maid",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "jveldt",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "bunq",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "tikkie",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "rijksoverheid",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "burned",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "j.veldt",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "shapeshift.io",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "changelly.com",
      },
    },
  },
  Object {
    "@@tag": "Just",
    "@@type": "folktale:Maybe",
    "@@value": Object {
      "value": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "changelly",
      },
    },
  },
]
`;
